packages/devtools_app/lib/src/shared/utils.dart

```diff
@@ -297,3 +298,69 @@ Future<void> launchUrlWithErrorHandling(String url) async {
     onError: () => notificationService.push('Unable to open $url.'),
   );
 }
+
+/// A worker that will run [callback] in groups of [chunkSize], when [doWork] is called.
+///
+/// [progressCallback] will be called with 0.0 progress when starting the work and any
+/// time a chunk finishes running, with a value that represents the proportion of
+/// indices that have been completed so far.
+///
+/// This class may be helpful when sets of work need to be done over a list, while
+/// avoiding blocking the UI thread.
+class InterruptableChunkWorker {
+  InterruptableChunkWorker({
+    int chunkSize = _defaultChunkSize,
+    required this.callback,
+    required this.progressCallback,
+  }) : _chunkSize = chunkSize;
+
+  static const _defaultChunkSize = 50;
+
+  final int _chunkSize;
+  int _workId = 0;
+  void Function(int) callback;
+  void Function(double progress) progressCallback;
+
+  /// Start doing the chunked work.
+  ///
+  /// [callback] will be called on every index from 0...[length-1], inclusive,
+  /// in chunks of [_chunkSize]
+  ///
+  /// If [doWork] is called again, then [callback] will no longer be called
+  /// on any remaining indices from previous [doWork] calls.
+  ///
+  Future<bool> doWork(int length) {
+    final completer = Completer<bool>();
+    final localWorkId = ++_workId;
+
+    void doChunkWork(int chunkStartingIndex) {
+      if (chunkStartingIndex >= length) {
+        return completer.complete(true);
+      }
+
+      final chunkUpperIndexLimit = min(length, chunkStartingIndex + _chunkSize);
+
+      for (int indexIterator = chunkStartingIndex;
+          indexIterator < chunkUpperIndexLimit;
+          indexIterator++) {
+        // If our localWorkId is no longer active, then do not continue working
+        if (localWorkId != _workId) return completer.complete(false);
+        callback(indexIterator);
+      }
+
+      progressCallback(chunkUpperIndexLimit / length);
+      Future.delayed(const Duration(), () {
```

Comment:
use Duration.zero, and add a comment explaining why we are delaying a zero duration.